README.md

```diff
@@ -48,61 +48,193 @@ When a job is available, the controller will create a pod to acquire and run the
 The entrypoint rewriting and ordering logic is heavily inspired by [the approach used in Tekton](https://github.com/tektoncd/pipeline/blob/933e4f667c19eaf0a18a19557f434dbabe20d063/docs/developers/README.md#entrypoint-rewriting-and-step-ordering).
 
 ## Architecture
-
 ```mermaid
 sequenceDiagram
-    participant bc as buildkite controller
-    participant gql as Buildkite GraphQL API
-    participant bapi as Buildkite API
-    participant kubernetes
-    bc->>gql: Get scheduled builds & jobs
-    gql-->>bc: {build: jobs: [{uuid: "abc"}]}
-    kubernetes->>pod: start
-    bc->>kubernetes: watch for pod completions
-    bc->>kubernetes: create pod with agent sidecar
-    kubernetes->>pod: create
-    pod->>bapi: agent accepts & starts job
-    pod->>pod: run sidecars
-    pod->>pod: agent bootstrap
-    pod->>pod: run user pods to completion
-    pod->>bapi: upload artifacts, exit code
-    pod->>pod: agent exit
-    kubernetes->>bc: pod completion event
-    bc->>kubernetes: cleanup finished pods
+    participant BC as Buildkite Controller
+    participant GQL as Buildkite GraphQL API
+    participant API as Buildkite API
+    participant K8S as Kubernetes
+    participant Pod
+
+    BC->>GQL: 1. Fetch scheduled builds & jobs
+    GQL-->>BC: 2. Respond with builds & jobs<br>{build: jobs: [{uuid: "abc"}]}
+    
+    BC->>K8S: 3. Create pod with agent sidecar
```

Comment:
Here controller actually asks for a k8s job to be created and then k8s will create a job and pod behind that job

Reply:
I only added numbers to the steps in this diagram to make it clearer but I believe thats whats shown in the diagram, the buildkite controller requests the job from kubernetes and then initializes the pod - is that right?